hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -722,7 +722,18 @@ private class Emit(
             ))))
 
       case _: ArrayMap | _: ArrayFilter | _: ArrayRange | _: ArrayFlatMap | _: ArrayScan | _: ArrayLeftJoinDistinct | _: ArrayAggScan | _: ReadPartition =>
-        emitArrayIterator(ir).toEmitTriplet(mb, PArray(coerce[PStreamable](ir.pType).elementType))
+        val m = mb.newField[Boolean]("arr_m")
```

Comment:
hmm… I think I'd prefer if you either used the `wrapToMethod` on the Emit class, which would look something like this:
```
wrapToMethod(Seq(x)) { (mb, ptype, et) => Code(et.setup, et.m.mux(m := true, Code(m := false, v := et.value))) }
```
or just constructed and invoked the method builder directly, since you're only trying to generate at most one method here. Most of this boilerplate is intended to dynamically group (and tier) code chunks based on expected size, which doesn't really come into play here.

Reply:
ah wait---the first thing wouldn't work because you want an emit iterator in there, not an emit triplet.